### PR TITLE
Merge flat/eager creates

### LINKED ISSUE #304: Flat/Eager creation need some work
- [x] Eager and Flat Create should be refactored into one function.
- [ ] After checking if nested model ids are zero we should also check if that model exists

### PR DESCRIPTION
**Pull request by mclark4386**
Should fix #304

### CODE REVIEW

**Comment by u007** [comment]
i think we might want to document the flow of how eager works, sometimes i'm also lost after awhile of not touching this part of the code

File: executors.go [hunk 1]
```diff
@@ -97,24 +94,80 @@ func (c *Connection) ValidateAndCreate(model interface{}, excludeColumns ...stri
...
-			asos, err := associations.ForStruct(m.Value)
+			var localIsEager = isEager
+			asos, err := associations.ForStruct(m.Value, c.eagerFields...)
 			if err != nil {
 				return err
 			}
 
+			if localIsEager {
+				// No association, fallback to non-eager mode.
+				if len(asos) == 0 {
```
Comment (stanislas-m) [inline executors.go]: How about combining the two conditions here?

File: executors.go [hunk 2]
```diff
@@ -133,6 +185,24 @@ func (c *Connection) Create(model interface{}, excludeColumns ...string) error {
 						continue
 					}
 
+					if localIsEager {
```
Comment (stanislas-m) [inline executors.go]: I'm not sure the `BeforeSetup` part should be outside the if (in non-eager cases, does it make sense to set the ID field from a struct we didn't touch?). If you confirm that, the whole `if` can be moved outside the loop.

File: executors.go [hunk 3]
```diff
@@ -157,6 +227,39 @@ func (c *Connection) Create(model interface{}, excludeColumns ...string) error {
...
+								}
+							}
+							return nil
+						})
+
+						if err != nil {
+							return err
+						}
+					}
 					stm := after[index].AfterProcess()
```
Comment (stanislas-m) [inline executors.go]: Same thing here: do we really have anything to execute in non-eager mode?

File: executors.go (hunk 2 above)
Comment (mclark4386) [reply executors.go]: I'm pretty sure those are how the m2m/belongs_to are getting set properly in the flat case.

**Pull request merged**